library/src/main/java/com/okta/appauth/android/AuthPayload.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2018, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License,
+ * Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the
+ * License.
+ */
+
+package com.okta.appauth.android;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * That is payload that is used in order to provide additional parameters
+ * for Authorization request or alter default parameters values (like mState).
+ *
+ */
+public class AuthPayload {
```

Comment:
We might want to name this something more oauth/oidc specific (or maybe a different sub package) to not confuse this and the authn-sdk?

Reply:
We can rename it as AuthAditionalPayload, but may be it is better to keep as a part of appauth package because it actually adds parameters for authentication request that are specified as a part of OpenId standard.